app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class LikesController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @like = Like.new(user_id: current_user.id, post_id: @post.id)
+    @like.user = current_user
+    @like.post = Post.find(params[:post_id])
+    @like.save
+    @like.update_like_counter
+    redirect_to user_post_path(current_user, @post.id)
+  end
```

Comment:
- [ ] When I add like, the like counter increases by 2, not 1. This, in my opinion, is because the `update_like_counter` method was used twice: once in the Like model (`after_save :update_like_counter`) and once in the create action controller (`@like.update_like_counter`). I recommend removing one.